### PR TITLE
Board defines defined

### PR DESCRIPTION
**Pull request by tsaxon**
Pin locations corresponding to our DSX lookup table are included in main by including board_defines.h. Not sure about definitions for user/debug LED's as well as SPI/I2C, but all have been initialized and occupy pins on the chip. Running out of pin space quickly.

ADC5 has been defined and uses 1 channel (connected to an external pin) to read current values and 1 internal channel (happening inside MCU somewhere) to read the temperature of the microcontroller. Haven't tested either functionality yet, just defined.

### CODE REVIEW

File: dsx-fw/DSX_Library/ADC/Inc/adc.h
Comment (manraj-s) [inline dsx-fw/DSX_Library/ADC/Inc/adc.h:16]: I think this variable should be define in the header file rather than the source file

**Comment by jaypacamarra** [comment]
I may be blind, but did you define the chip enable (CE) / slave select (SS) pin for SPI communication?

File: dsx-fw/DSX_Library/Board_Defines/Inc/board_defines.h
Comment (jaypacamarra) [inline dsx-fw/DSX_Library/Board_Defines/Inc/board_defines.h]: These don't need to go in board defines.

File: dsx-fw/DSX_Library/Board_Defines/Inc/board_defines.h
Comment (jaypacamarra) [inline dsx-fw/DSX_Library/Board_Defines/Inc/board_defines.h]: These as well don't need to be in board defines. The reason is that these #defines in the board_defines.h file will be used to verify if the loc is correct. The user won't have a direct command in the lookup table to modify the Debug LEDS, I2C pins, and SPI pins.

**Comment by jaypacamarra** [comment]
Also, sorry for adding this late. I also need 7 GPIO output pins defined called:

- MUX1_S0
- MUX1_S1
- MUX1_S2
- MUX1_S3
- MUX2_S0
- MUX3_S0
- MUX_En

Mux 1 is going to be a 16 channel analog mux and mux 2 and mux 3 is a 2 channel analog mux. I'll use them to sense current on all of our user pins. These 6 pins will help to select which channel we want to output on the common pin of the mux. The MUX_En is an enable pin which ill use in case of an over current which should help cutoff any harmful over current continuously going into the MCU. See the [current sensing functional block diagram](https://drive.google.com/file/d/1TCTKODYVUE81bcNQ2N0TZDmDBHErQa0a/view?usp=sharing).

**Comment by jaypacamarra** [comment]
@Harjaapb are we missing anything?

File: dsx-fw/DSX_Library/Board_Defines/Inc/board_defines.h
Comment (jaypacamarra) [inline dsx-fw/DSX_Library/Board_Defines/Inc/board_defines.h:46]: Could we change the USER_LD2 as the front panel fault LED? It goes red if there's a fault like over current. One user LED should be enough for the user.

Comment (jaypacamarra) [inline dsx-fw/DSX_Library/Board_Defines/Inc/board_defines.h:46]: Could we change:
``` C
#define USER_LD2 1U
to
#define FAULT_LD2 1U  // or something like that
```
to a front panel fault LED for the purpose of turning RED if a fault is detected?

**Review by Harjaapb — APPROVED** [review]
Agree with the comments above

**Comment by jaypacamarra** [comment]
@Harjaapb You shouldn't approve a pull request if no changes or comments have been made that addresses those comments. Right now, we can't approve this pull request yet.

File: dsx-fw/DSX_Library/Board_Defines/Inc/board_defines.h
Comment (tsaxon) [reply dsx-fw/DSX_Library/Board_Defines/Inc/board_defines.h]: Makes sense, wasn't sure so threw them in anyways. Will take them out and see what we can do about the mux pins

**Comment by tsaxon** [comment]
Not sure about the extra SPI pins, all that is defined is what CubeMX initialized for the SPI. Have never used SPI, maybe full-duplex master isn't what we want?

**Comment by jaypacamarra** [comment]
Full-duplex master should be good. Also @tsaxon I'm gonna check out your changes tonight and hopefully I can merge it as well.

**Comment by jaypacamarra** [comment]
Looks good! I'm gonna merge.

**Pull request merged**
